contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -26,6 +27,10 @@ contract LockDealNFT is LockDealNFTModifiers, ILockDealNFTEvents {
         token = vaultManager.vaultIdToTokenAddress(poolIdToVaultId[poolId]);
     }
 
+    function providerOf(uint256 poolId) external view returns (IProvider provider) {
```

Comment:
After adding these changes, we will have two points to call the **Provider** by id. Using the default `poolIdToProvider` and this case. But I like the idea of simplifying the name and returning a direct `IProvider` type. We can move this function and other getters to the `LockDealNFTState` file.